packages/kolibri-components/src/KLabeledIcon.vue

```diff
@@ -40,6 +40,24 @@
         required: false,
       },
     },
+    computed: {
+      labelEmpty() {
+        if (this.label) {
+          return false;
+        }
+
+        if (!('default' in this.$slots) || !this.$slots.default.length) {
+          return true;
+        }
+
+        const defaultSlot = this.$slots.default[0];
+        return !(
+          defaultSlot.text ||
+          defaultSlot.tag ||
+          (defaultSlot.children && defaultSlot.children.length)
+        );
+      },
+    },
```

Comment:
I think this implementation could have been a bit simpler. Something like:

template:

```html
<slot>{{ computedLabel }}</slot>
```

computed prop:

```js
computedLabel() {
  const zeroWidth = "\u200b";
  return this.label ? this.label : zeroWidth;
},
```

Reply:
Unfortunately I don't think that works for *empty* passed into the slot. I'm pretty sure I tried it. Those two different methods, label vs slot, made this more tricky than I though.

Reply:
I don't think there's any usage of KLabledIcon with nothing in the default slot. I think if somebody tries to use this component without a label prop or anything in the slot, we could log a warning since it's probably a mistaken usage.

Reply:
Well, for clarification, if `someVar` in my example below turns out to be empty, Devon's suggestion would not end up working.
```
<KLabeledIcon>{{someVar}}</KLabeledIcon>
```
So I wanted this to be consist with an empty variable was passed to the `label` prop.